frontend/cypress/tests/codeView/codeView_switchAuthorship.js

```diff
@@ -0,0 +1,93 @@
+describe('switch authorship', () => {
+  it('switch authorship view should restore all default controls', () => {
+    Cypress.wait();
+
+    // open the code panel
+    cy.get('.icon-button.fa-code')
+        .should('be.visible')
+        .first()
+        .click();
+
+    // change sort by
+    cy.get('#tab-authorship > .title > .contribution > .sorting > .sort-by > select')
+        .select('Path')
+        .should('have.value', 'path');
+
+    // change sort order
+    cy.get('#tab-authorship > .title > .contribution > .sorting > .sort-order > select')
+        .select('Ascending')
+        .should('have.value', 'false');
+
+    // uncheck a file type
+    cy.get('#tab-authorship > .title > .contribution > .fileTypes input[id="gradle"]')
+        .uncheck()
+        .should('not.be.checked');
+
+    // switch authorship view
+    cy.get('.icon-button.fa-code')
+        .should('be.visible')
+        .last()
+        .click();
+
+    // check default controls
+    cy.get('#tab-authorship > .title > .contribution > .sorting > .sort-by > select')
+        .should('not.have.value', 'path')
+        .should('have.value', 'lineOfCode');
+
+    cy.get('#tab-authorship > .title > .contribution > .sorting > .sort-order > select')
+        .should('have.value', 'true');
+
+    cy.get('#tab-authorship > .title > .contribution > .fileTypes > .radio-button--checkbox')
+        .should('be.checked');
+
+    cy.get('#tab-authorship > .title > .contribution > .fileTypes input[id="all"]')
+        .should('be.checked');
+
+    cy.get('#tab-authorship > .title > .contribution > .fileTypes input[id="gradle"]')
+        .should('be.checked');
+
+    cy.get('#tab-authorship > .title > .contribution > .fileTypes input[id="java"]')
+        .should('be.checked');
+
+    cy.get('#tab-authorship > .title > .contribution > .fileTypes input[id="yml"]')
+        .should('be.checked');
+  });
+
+  it('switch authorship view should not retain information from previous visited tabs', () => {
+    Cypress.wait();
+
+    // open the code panel
+    cy.get('.icon-button.fa-code')
+        .should('be.visible')
+        .first()
+        .click();
+
+    cy.get('#tab-authorship .panel-heading .author span')
+        .should('contain.text', 'eugenepeh');
```

Comment:
The implementation here is assuming that eugenepeh will be the first in the group. However, this may not always be true as we might edit the configuration (add author, etc.). Could you change this part of the test to allow it to handle dynamic configuration?

Reply:
I think it is hard to handle dynamic configuration as we at least need a guarantee that the first author in the repos we switch from and to is not the same. Otherwise there is nothing to test.